src/PrivateDump/Transformer.php

```diff
@@ -133,6 +165,25 @@ public function transform($value, $replacement)
             return $replacement;
         }
 
+
+        $object = null;
+        $matches = null;
+        if (preg_match('/^@(\w+)\((\w+\))\.(.*)$/', $replacement, $matches)) {
+            $objectType = $matches[1];
+            $objectName = $matches[2];
+
+            if (array_key_exists($objectName, $this->objectCache)) {
+                $object = $this->objectCache[$objectName];
+            } else {
+                $getObjectMethod = sprintf('transformObject%s', $objectType);
+                if (method_exists($this, $getObjectMethod)) {
+                    $object = $this->$getObjectMethod();
+                    $this->objectCache[$objectName] = $object;
+                }
+            }
+            $replacement = "@{$matches[3]}";
```

Comment:
Really hard at a glance to tell what any of this is actually doing, could use either some nicely named methods, or comments

Reply:
Added some comments, let me know if enough, otherwise I'll refactor.